### PR TITLE
[ci][core] Add more visbility into state api stress test 

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?
The workload seems to be stuck while creating objects - it's not part of what the test aims to test. 

I wonder if there's some regression or new issues that's orthogonal to what this test is doing. 

I am also unstabling this test for now while I look into the failures. (The test has become unstable recently, with failures that are not relevant to the state API) 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/release_tests.yaml [hunk 1]
```diff
@@ -4790,6 +4790,7 @@
 - name: stress_test_state_api_scale
   group: core-daily-test
   working_dir: nightly_tests
+  stable: false
```
Comment (jjyao) [inline release/release_tests.yaml:4793]: I think we now use jailed?

**Review by jjyao — APPROVED** [review]

File: release/ray_release/scripts/run_release_test.py [hunk 2]
```diff
@@ -167,6 +167,9 @@ def main(
             no_terminate=no_terminate,
         )
         return_code = result.return_code
+        from rich import print
+
+        print(result)
```
Comment (aslonnie) [inline release/ray_release/scripts/run_release_test.py:172]: remove these? why these lines are required?

Comment (rickyyx) [reply release/ray_release/scripts/run_release_test.py:172]: oops - some unrelated changes. good call.

File: release/nightly_tests/stress_tests/test_state_api_scale.py [hunk 3]
```diff
@@ -165,7 +170,7 @@ def exit(self):
 
 def test_many_objects(num_objects, num_actors):
     if num_objects == 0:
-        print("Skipping test with no objects")
+        logger.info("Skipping test with no objects")
```
Comment (rickyyx) [inline release/nightly_tests/stress_tests/test_state_api_scale.py:173]: so we have timestamps.

**Comment by rickyyx** [comment]
Running a test here: https://buildkite.com/ray-project/release-tests-pr/builds/42381

**Comment by rickyyx** [comment]
Passed: https://buildkite.com/ray-project/release-tests-pr/builds/42462#0188c51b-2037-4583-970d-960950ec2dfd

**Pull request merged**